server/src/main/java/org/elasticsearch/index/query/GeoShapeQueryBuilder.java

```diff
@@ -383,27 +393,86 @@ protected Query doToQuery(QueryShardContext context) {
+                ShapeRelation.CONTAINS + " query relation not supported for Field [" + fieldName
+                    + "] when using Strategy [" + shapeFieldType.strategy() + "]");
+        }
+
+        // wrap geoQuery as a ConstantScoreQuery
+        return new ConstantScoreQuery(getVectorQuery(context, queryShapeBuilder.buildLucene()));
+    }
+
+    private Query getVectorQuery(QueryShardContext context, Object queryShape) {
+        Query geoQuery;
+        if (queryShape instanceof Line[]) {
+            geoQuery = LatLonShape.newLineQuery(fieldName(), relation.getLuceneRelation(), (Line[]) queryShape);
+        } else if (queryShape instanceof Polygon[]) {
+            geoQuery = LatLonShape.newPolygonQuery(fieldName(), relation.getLuceneRelation(), (Polygon[]) queryShape);
+        } else if (queryShape instanceof Line) {
+            geoQuery = LatLonShape.newLineQuery(fieldName(), relation.getLuceneRelation(), (Line) queryShape);
+        } else if (queryShape instanceof Polygon) {
+            geoQuery = LatLonShape.newPolygonQuery(fieldName(), relation.getLuceneRelation(), (Polygon) queryShape);
+        } else if (queryShape instanceof Rectangle) {
+            Rectangle r = (Rectangle) queryShape;
+            geoQuery = LatLonShape.newBoxQuery(fieldName(), relation.getLuceneRelation(),
+                r.minLat, r.maxLat, r.minLon, r.maxLon);
+        } else if (queryShape instanceof double[][]) {
+            // note: we decompose point queries into a bounding box query with min values == max values
+            // to do this for multipoint we would have to create a BooleanQuery for each point
+            // this is *way* too costly. So we do not allow multipoint queries
+            throw new QueryShardException(context, "Field [" + fieldName + "] does not support " + GeoShapeType.MULTIPOINT + " queries");
+        } else if (queryShape instanceof double[] || queryShape instanceof GeoPoint) {
+            // for now just create a single bounding box query with min values == max values
+            double[] pt = queryShape instanceof GeoPoint
+                ? new double[] {((GeoPoint)queryShape).lon(), ((GeoPoint)queryShape).lat()}
+                : (double[])queryShape;
+            return LatLonShape.newBoxQuery(fieldName, relation.getLuceneRelation(), pt[1], pt[1], pt[0], pt[0]);
+        } else if (queryShape instanceof Object[]) {
+            geoQuery = createGeometryCollectionQuery(context, (Object[]) queryShape);
+        } else {
+            throw new QueryShardException(context, "Field [" + fieldName + "] found and unknown shape");
+        }
+        return geoQuery;
+    }
+
+    private Query createGeometryCollectionQuery(QueryShardContext context, Object... shapes) {
+        BooleanQuery.Builder bqb = new BooleanQuery.Builder();
+        for (Object shape : shapes) {
+            bqb.add(getVectorQuery(context, shape), BooleanClause.Occur.SHOULD);
+        }
+        return bqb.build();
```

Comment:
let's wrap in a ConstantScoreQuery?

Reply:
Its wrapped in a `ConstantScoreQuery` on line 432.